.github/workflows/staging.yml

```diff
@@ -0,0 +1,159 @@
+name: Staging Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Release tag (e.g. v4.4.0)'
+        required: true
+        type: string
+
+env:
+  TAG: ${{ inputs.tag }}
+
+jobs:
+  image-build-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+
+      - uses: cachix/install-nix-action@v31.3.0
+
+      - name: Pre-populate nix-shell
+        run: |
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --run "echo" shell.nix
+
+      - name: Set Image Org
+        run: |
+          [ -z "${{ secrets.IMAGE_ORG }}" ] && IMAGE_ORG=openebs || IMAGE_ORG=${{ secrets.IMAGE_ORG }}
+          echo "IMAGE_ORG=${IMAGE_ORG}" >> $GITHUB_ENV
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push the release images
+        run: |
+          nix-shell --pure --run "./scripts/helm/update-version-upon-release.sh"
+          nix-shell --run "./scripts/release.sh --tag ${TAG} --registry ghcr.io/${{ env.IMAGE_ORG }}/dev"
+
+  kubectl-plugin:
```

Comment:
let's not duplicate both the image build and the kubectl plugin build
Ex: we could either dispatch the other workflow and collect its result or just dispatch it and do the work there